### ISSUE TITLE
Can't seek in audio files when using lavfi-complex

### ISSUE THREAD

**Issue by kokoko3k** [mpv-player/mpv#3688]
### mpv version and platform

0.21
### Reproduction steps

try to seek with the following:

```
mpv file.mp3 --lavfi-complex='[aid1]asplit[ao][a1];[a1] avectorscope=m=lissajous_xy:s=256x256:r=25 [spectrum];[vid1][spectrum]overlay[vo]'
```
### Expected behavior

immediate seek
### Actual behavior

audio stops and resumes after seconds
### Log file

http://pastebin.com/XG8vm02z

Bisected, the bad commit is:

```
# git bisect bad
3a5cbf39072fd07f457560f25899dda442246b20 is the first bad commit
commit 3a5cbf39072fd07f457560f25899dda442246b20
Author: wm4 <wm4@nowhere>
Date:   Sun Oct 2 13:47:19 2016 +0200

    audio: fix late audio start

    Regression since commit bbcd0b6a. This code is just cursed, because it's
    a fragile state machine with no proper tests, and which could be done in
    a much simpler way. Without doubt this change will cause a regression in
    some ridiculous corner case as well.

    Fixes #3610 (the cause of it, not the behavior it resulted in).
```

hr-seek works.

**Comment by wm4** [comment]
Can't reproduce.

**Comment by kokoko3k** [comment]
Wops sorry, wrong filter, please try this one:
``
mpv /tmp/file.mp3 --lavfi-complex='[aid1]asplit=3 [azero][auno][ao];[azero]showspectrum=mode=separate:color=intensity:slide=scroll:s=4x512:color=intensity:win_func=blackman:mode=combined:saturation=1,crop=h=(in_h/1.5):y=in_h,scale=h=512:w=512:force_original_aspect_ratio=disable[spectrum];[auno]showwaves=s=512x512:mode=line:r=20,crop=h=(ih/2):y=(0),crop=h=(ih/2):y=ih,colorlevels=romax=0.5:bomin=1:gomin=1[waves];movie=/tmp/gambas.0/15042/2.tmp,scale=h=(128):w=512[pic];[spectrum][pic]overlay[overlay1];[overlay1][waves]overlay[vo]'
```

**Comment by kokoko3k** [comment]
Yep, sorry, that line only works with files that includes album-art.
please, try this one:
[code]mpv --hr-seek=no /tmp/file.mp3 --lavfi-complex='[aid1]asplit[ao][a1];[a1] showspectrum [vo]'[/code]

I just noticed that not all of the filters trigger the issue; showspectrum does.

**Comment by wm4** [comment]
Still can't reproduce, and yes I tried the one before with a file that has album art.

**Comment by kokoko3k** [comment]
Mmh, what could it be?
My ffmpeg version is 3.1.4 package from Arch Linux.
I'll try to build it myself.

**Comment by kokoko3k** [comment]
I rebuilded mpv from master and ffmpeg from master, but unfortunately the issue is still present.
I don't think is relevant, however i've uploaded the mp3 file i'm testing (i tested others too) here:
http://wpage.unina.it/aorefice/sharevari/file.mp3

**Comment by wm4** [comment]
That file behaves indeed weird with your first test, because "[vid1]" will be EOF, but works with the second one.

**Comment by kokoko3k** [comment]
Yes, i expected that it wouldn't work with the first example, as it lacks any "video", what is weird to me is that it works for you with the second example.
i just tried with --msg-level=all=trace, but still, while the playback stops, the log is completely frozen too.
I produced another log with trace details.
There is a note in there i wrote to mark when the player was frozen.
http://pastebin.com/hLxcT9Ti

**Comment by wm4** [comment]
Did you try with ffmpeg git master?

**Comment by kokoko3k** [comment]
Yes, yes, i tried both mpv and ffmpeg from git master

**Comment by kokoko3k** [comment]
Going on with the strangeness with the issue, if i add --video-sync=display-* or --video-sync=desync the issue goes away.
I'm wondering if by any chance you have video-sync=display-* in the profile you used to test (?)

**Comment by wm4** [comment]
Could reproduce. It's because the ffmpeg showspectrum filter returns a video PTS of 0 as first frame even though the audio input PTS isn't even close to 0, and only the second frame PTS is based on the audio input PTS again.

**Comment by lachs0r** [comment]
This appears to be fixed in at least FFmpeg 3.3.2.
Maybe it was this: FFmpeg/FFmpeg@49d0678181af67455dfd23b98ab3857db1b68e67
